spec.html

```diff
@@ -21167,6 +21165,23 @@ <h1>Runtime Semantics: ClassDefinitionEvaluation</h1>
           1. Perform _classScope_.InitializeBinding(_classBinding_, _F_).
         1. Return _F_.
       </emu-alg>
+
+      <emu-clause id="sec-default-constructor-functions">
+        <h1>Default Constructor Functions</h1>
+        <p>When a Default Constructor Function is called with zero or more arguments with form the rest parameter ..._args_, the following steps are taken:</p>
+        <emu-alg>
+          1. If NewTarget is *undefined*, throw a *TypeError* exception.
+          1. If _F_.[[ConstructorKind]] is ~derived~, then
+            1. NOTE: This branch behaves similarly to `constructor(...args) { super(...args); }`.
+            1. Let _F_ be the active function object.
+            1. Let _func_ be ? _F_.[[GetPrototypeOf]]().
+            1. If IsConstructor(_func_) is *false*, throw a *TypeError* exception.
+            1. Return ? Construct(_func_, _args_, NewTarget).
+          1. Else,
+            1. NOTE: this branch behaves similarly to `constructor() {}`.
```

Comment:
similarly, or identically?

Reply:
In the context of the spec, similarly. If it were identical it would also set up a lexical this environment, which it does not. For js end users, it should be identical.

Reply:
I think for the purposes of the spec, this does not perform the same operations as an ecmascript function `constructor() {}`, even if it has the same outward behaviour, so saying identically would not be correct. For example, you don't have to worry about your implementation of `this` environments while implementing this.